### PR TITLE
Alter license

### LINKED ISSUE #320: License change
AGPL licenses can be a pain for many users, how do we feel about moving towards a more permissive license (e.g. Apache 2.0)? Given that pymagicc wraps a non-commercial binary, there's no real commercial worries anywhere and the fix in licensing really needs to come from MAGICC (we're working on that, a proposal was recently accepted to take it open-source).

@rgieseke @lewisjared @swillner @matthiasmengel any objections (also hi!)?

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Closes #320 

Please confirm that this pull request has done the following:

- [x] Tests added (N/A)
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable) (N/A)
- [x] Description in ``CHANGELOG.rst`` added

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -350,8 +350,7 @@ See also the `MAGICC website <http://magicc.org/>`_ and
 `Wiki <http://wiki.magicc.org/index.php?title=Main_Page>`_
 for further information.
 
-The ``pymagicc`` wrapper is free software under the GNU Affero General Public
-License v3, see `LICENSE <./LICENSE>`_.
+The ``pymagicc`` wrapper is free software. For license details, see `LICENSE <./LICENSE>`_.
```
Comment (rgieseke) [inline README.rst:353]:
```suggestion
The ``pymagicc`` wrapper itself is licensed under a BSD-3 license. For details, see `LICENSE <./LICENSE>`_.
```
"Free" software is historically understood to be different from "Open Source" software.
https://en.wikipedia.org/wiki/Free_and_open-source_software

File: pymagicc/__init__.py [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Copyright (c) 2017 pymagicc authors:
-#   Robert Gieseke <robert.gieseke@pik-potsdam.de>
-# Free software under GNU Affero General Public License v3, see LICENSE
+#   - Robert Gieseke <robert.gieseke@pik-potsdam.de>
```
Comment (rgieseke) [inline pymagicc/__init__.py:2]: Let's do it the Pandas way maybe?
```
Copyright (c) 2008-2011, AQR Capital Management, LLC, Lambda Foundry, Inc. and PyData Development Team
Copyright (c) 2011-2020, Open source contributors.
```
https://github.com/pandas-dev/pandas/blob/master/LICENSE

```
 Pymagicc developers and contributors
```

I think it's not even necessary to have this in the code file, Pandas hasn't either.

File: pymagicc/__init__.py [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
+#   - Zeb Nicholls <zebedee.nicholls@climate-energy-college.org>
+#   - Jared Lewis <jared.lewis@climate-energy-college.org>
+#   - Sven Willner <sven.willner@pik-potsdam.de>
+#   - Matthias Mengel <matthias.mengel@pik-potsdam.de>
+#
+# Free software. For license details, see LICENSE.
```
Comment (rgieseke) [inline pymagicc/__init__.py:8]:
```suggestion
# For license details, see LICENSE.
```
Not "free" software anymore ... :smile_cat: Still open of course ...

File: LICENSE [hunk 4]
```diff
@@ -1,661 +1,14 @@
-                    GNU AFFERO GENERAL PUBLIC LICENSE
-                       Version 3, 19 November 2007
+BSD 3-Clause License
 
- Copyright (C) 2007 Free Software Foundation, Inc. <http://fsf.org/>
- Everyone is permitted to copy and distribute verbatim copies
- of this license document, but changing it is not allowed.
+Copyright (c) 2020, Robert Gieseke, Zebedee Nicholls, Jared Lewis, Sven Willner, Matthias Mengel
```
Comment (rgieseke) [inline LICENSE:3]:
```suggestion
Copyright (C), Pymagicc developers and contributors
```
Maybe add a "see setup.py" for the full list?

File: LICENSE [hunk 5]
```diff
@@ -1,661 +1,14 @@
...
+All rights reserved.
```
Comment (rgieseke) [inline LICENSE:4]:
```suggestion
```
I'd remove this as it makes unclear and is not necessary.
https://opensource.stackexchange.com/questions/2121/mit-license-and-all-rights-reserved

File: README.rst (hunk 1 above)
Comment (znicholls) [reply README.rst:353]: Ah ok. Shall we add free too then?

Comment (znicholls) [reply README.rst:353]:
```suggestion
The ``pymagicc`` wrapper itself is free software released under a BSD-3 license. For license details, see `LICENSE <./LICENSE>`_.
```

File: pymagicc/__init__.py (hunk 2 above)
Comment (znicholls) [reply pymagicc/__init__.py]: Updated in c37c7b2

File: LICENSE (hunk 4 above)
Comment (znicholls) [reply LICENSE]: Fixed by ec01a5

File: LICENSE (hunk 5 above)
Comment (znicholls) [reply LICENSE]: Resolved in 1d12d629

File: pymagicc/__init__.py (hunk 3 above)
Comment (znicholls) [reply pymagicc/__init__.py]: I've removed these comments anyway because pandas doesn't have anything like this anyway. However, how is it not free anymore?

File: README.rst (hunk 1 above)
Comment (rgieseke) [reply README.rst:353]: Technically, BSD is free software but i think it's confusing as i myself was confused and thought "free" only applied to GPL-style software. I think people even further away from these FOSS dialectics (like potential scientist users) and the "free as in speech or free as in beer" differences i'd still leave "free" out probably. (Or write "open source")

File: pymagicc/__init__.py (hunk 3 above)
Comment (rgieseke) [reply pymagicc/__init__.py]: See above, i was confused ...

Comment (rgieseke) [reply pymagicc/__init__.py]: I think the politics of "free software" are especially tricky in this repo as the binary is clearly not "free software". It's written that they have different licenses but it's still confusing ...

File: README.rst (hunk 1 above)
Comment (znicholls) [reply README.rst:353]: ok easy

File: pymagicc/__init__.py (hunk 3 above)
Comment (znicholls) [reply pymagicc/__init__.py]: Ok all good

**Comment by rgieseke** [comment]
I suggested

> Maybe add a "see setup.py" for the full list?

but am wondering whether that's actually a good solution. One probably won't list every typo-fixing one-time contributor there.
One could use a link to https://github.com/openscm/pymagicc/graphs/contributors ?

```
Pymagicc developers, see also https://github.com/openscm/pymagicc/graphs/contributors
```

Thoughts?

**Comment by znicholls** [comment]
Also looks good, I’m happy with whatever

File: LICENSE [hunk 6]
```diff
@@ -1,661 +1,13 @@
-                    GNU AFFERO GENERAL PUBLIC LICENSE
-                       Version 3, 19 November 2007
+BSD 3-Clause License
 
- Copyright (C) 2007 Free Software Foundation, Inc. <http://fsf.org/>
- Everyone is permitted to copy and distribute verbatim copies
- of this license document, but changing it is not allowed.
+Copyright (c) 2017-2021, Pymagicc developers and contributors, see `setup.py` for full list.
```
Comment (rgieseke) [inline LICENSE:3]:
```suggestion
Copyright (c) 2017-2021, Pymagicc developers, see also https://github.com/openscm/pymagicc/graphs/contributors
```

**Comment by rgieseke** [comment]
> Also looks good, I’m happy with whatever

Cool, just trying to avoid writing down workflows which can become hard to follow in practice ...

**Review by rgieseke — APPROVED** [review]

**Comment by rgieseke** [comment]
Maybe also take the opportunity to cut a 2.0 release? ;-)
If you're worried about breaking changes we can always switch to Browser-like version numbering :-)

**Comment by znicholls** [comment]
> we can always switch to Browser-like version numbering

haha no that's a nightmare for users

**Comment by znicholls** [comment]
> Maybe also take the opportunity to cut a 2.0 release?

Yes will make a PR
